portfolio/src/main/java/com/google/sps/servlets/ChartServlet.java

```diff
@@ -26,26 +26,37 @@
 @WebServlet("/chart-data")
 public class ChartServlet extends HttpServlet {
 
+  private static final String JSON_TYPE = "application/json";
+  private static final String DATA_FILE = "/WEB-INF/edible_food_2011.csv";
+  private static final String UNDEFINED_DATA = "*";
+  private static final String FILE_DELIM = ",";
+  private static final int COUNTRY_INDEX = 1;
+  private static final int RICE_INDEX = 4;
+
   // Stores data in the form "country_name: rice_produced."
   private LinkedHashMap<String, Double> countryMap = new LinkedHashMap<>();
 
   @Override
   public void init() {
-    Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/edible_food_2011.csv"));
-
-    String headerLine1 = scanner.nextLine();
-    String headerLine2 = scanner.nextLine();
-    String headerLine3 = scanner.nextLine();
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream(DATA_FILE));
 
     while (scanner.hasNextLine()) {
       String line = scanner.nextLine();
-      String[] cells = line.split(",");
+      String[] cells = line.split(FILE_DELIM);
+
+      if (cells.length < RICE_INDEX + 1) {
+        continue;
+      } 
 
-      String country = String.valueOf(cells[1]);
+      String country = String.valueOf(cells[COUNTRY_INDEX]);
+      String riceStr = String.valueOf(cells[RICE_INDEX]);
+
+      if (country.length() == 0 || riceStr.length() == 0) {
```

Comment:
I would use isEmpty here instead of length, which is shorthand:
https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#isEmpty()
```suggestion
      if (country.isEmpty() || riceStr.isEmpty()) {
```